src/libvirtApi/domain.js

```diff
@@ -1089,3 +1089,29 @@ export async function domainAddTPM({ connectionName, vmName }) {
     const args = ["virt-xml", "-c", `qemu:///${connectionName}`, "--add-device", "--tpm", "default", vmName];
     return cockpit.spawn(args, { err: "message", superuser: connectionName === "system" ? "try" : null });
 }
+
+export function domainAttachVnc({ connectionName, vmName, vncAddress, vncPort, vncPassword }) {
+    const args = ['virt-xml', '-c', `qemu:///${connectionName}`, vmName, '--add-device', '--graphics', `vnc,listen=${vncAddress},port=${vncPort},passwd=${vncPassword}`];
+    const options = { err: "message" };
+
+    if (connectionName === "system")
+        options.superuser = "try";
+
+    return cockpit.spawn(args, options);
+}
+
+export function domainChangeVncSettings({
```

Comment:
This almost does the same thing as domainAttachVnc(), just `--add` vs. `--edit`. This could become a flag. But this is not important at all, please feel free to just resolve this.

Reply:
It also might have quoting issues, see domainSetDescription. I'll give this a polish.

Reply:
I have replaced this with our generalized virt-xml wrapper.  Using funny characters for password now works better, but virt-xml itself seems to have trouble. I'll file upstream bugs.

Reply:
https://github.com/virt-manager/virt-manager/issues/838